shared/src/telemetry/bgworker.rs

```diff
@@ -0,0 +1,127 @@
+    let extension_name = extension.name();
+    GucRegistry::define_bool_guc(
+        &format!("paradedb.{extension_name}.telemetry"),
+        &format!("Enable telemetry on the ParadeDB {extension_name} extension.",),
+        &format!("Enable telemetry on the ParadeDB {extension_name} extension.",),
+        &TELEMETRY_ENABLED,
+        GucContext::Userset,
+        GucFlags::default(),
+    );
+
+    // A background worker to read and send telemetry data to PostHog.
+    BackgroundWorkerBuilder::new(&format!("{}_telemetry_worker", extension.name()))
+        // Must be the name of a function in this file.
+        .set_function("telemetry_worker")
+        // Must be the name of the extension it will be loaded from.
+        .set_library(&extension.name())
+        // We pass the extension name to retrieve the associated data directory to read telemetry data from.
+        .set_argument((extension as i32).into_datum())
+        // Necessary for using plog!.
+        // Also, it doesn't seem like bgworkers will start without this.
+        .enable_spi_access()
+        // RecoveryFinished is the last available stage for bgworker startup.
+        // We wait until as late as possible so that we can make sure the
+        // paradedb.logs table is created, for the sake of using plog!.
+        .set_start_time(bgworkers::BgWorkerStartTime::RecoveryFinished)
+        .load();
+}
+
+#[pg_guard]
+#[no_mangle]
+#[allow(clippy::missing_safety_doc)]
+pub unsafe extern "C" fn telemetry_worker(extension_name_datum: pg_sys::Datum) {
+    let extension_i32 = unsafe { i32::from_datum(extension_name_datum, false) }
+        .expect("extension enum i32 not passed to bgworker");
+    let extension = ParadeExtension::from_i32(extension_i32)
+        .unwrap_or_else(|| panic!("unexpected extension i32 passed to bgworker {extension_i32}"));
+    let extension_name = extension.name();
+
+    pgrx::log!(
+        "starting {extension_name} telemetry worker at PID {}",
+        process::id()
+    );
+
+    // These are the signals we want to receive. If we don't attach the SIGTERM handler, then
+    // we'll never be able to exit via an external notification.
+    let sigterm_handler = SigtermHandler::new();
+    let settings_store = Box::new(GucSettings::new());
+    let telemetry_store = match PosthogStore::from_env().map(Box::new) {
+        Ok(store) => store,
+        Err(err) => {
+            pgrx::warning!("could not initialize posthog, exiting telemetry worker: {err}");
+            return;
+        }
+    };
+    let directory_store = PostgresDirectoryStore::new(&extension_name)
+        .map(Box::new)
+        .unwrap();
+    let sender = TelemetrySender {
+        extension_name: extension_name.to_string(),
+        telemetry_store,
+        directory_store,
+        settings_store,
+    };
+    let controller = TelemetryController {
+        sender,
+        directory_check_interval: Duration::from_secs(12 * 3600), // 12 hours
```

Comment:
Maybe we can make this a const somewhere? I suspect we might want to update this eventually

Reply:
Something Timescale does, for instance, is they send telemetry every hour for the first 12 hours, and then send it every 12hours. I believe they do this to catch people who churn quickly and try to understand why. I believe we should do this too, whether in this PR or a separate one

Reply:
I can add this behavior, but I'm not really sure we're collecting any data right now that would really help us "understand why". Do you think we're getting anything helping by sending the directory size check every hour?

Reply:
Let's keep it at what it is right now, you're right. We can improve our telemetry down the line